hazelcast/src/main/java/com/hazelcast/map/impl/MapServiceContext.java

```diff
@@ -190,4 +192,17 @@
     String addLocalListenerAdapter(ListenerAdapter listenerAdaptor, String mapName);
 
     IndexCopyBehavior getIndexCopyBehavior();
+
+    /**
+     * Returns the collection of the {@link RecordStoreMutationObserver}s
+     * for the given map's partition that need to be added in record
+     * store construction time in order to ensure no {@link RecordStore}
+     * mutations are missed.
+     *
+     * @param mapName The name of the map
+     * @param partitionId The partition
+     *
+     * @return The collection of the observers
+     */
+    Collection<RecordStoreMutationObserver<Record>> getRecordStoreMutationObservers(String mapName, int partitionId);
```

Comment:
Probably a better name is `createRecordStoreMutationObservers` since the implementations are always creating them.

Reply:
Renamed.